### PR TITLE
lgc minor tidyup

### PR DESCRIPTION
**Pull request by trenouf**
1927384e7 lgc cmake: change LLPC_BUILD_NAVI21 default to ON for standalone lgc build
7657dcefc lgc: restore built-ins list to alphabetical order
852dd3cd2 lgc: Minor comment improvements
f0ca53e31 cmake: Make lgc's interface directory public

### CODE REVIEW

File: lgc/CMakeLists.txt [hunk 1]
```diff
@@ -95,8 +95,10 @@ if(WIN32)
 endif()
 
 target_include_directories(LLVMlgc
+PUBLIC
```
Comment (dstutt) [inline lgc/CMakeLists.txt:98]: I don't understand what this bit is doing - but based on your comments about making the interface public I guess this makes sense if it works when you tested it.

**Review by dstutt — APPROVED** [review]
LGTM (with caveat that I don't know what the cmake interface change is doing)

**Comment by trenouf** [comment]
On Jonathan's suggestion:
V2: Remove LLPC_BUILD_NAVI21 as it is now always built.

**Comment by amdrexu** [comment]
@trenouf One of the commits could resolve an internal build error. You can merge this PR once the testing is finished. Thanks.

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 733083bdaec4ca2e66c2f68d69346671237dc6aa
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 1e61f9ab6eb2041c73d7e7bd8f2021fa05a4bdd8</li>
<li>SPVGEN: dev 7c1cf34dd7eb4d3a6e6652c26292eef1bfcbcaf0</li>
<li>XGL: dev 2e6b9ee239ce2435550441fe08dd511ed0c0642f</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev bcd2fe473ee79c4907f2136ab963c1a3676c9db5</li>
</ul></details>

<details><summary>CTS tests (Failed: 336/224701)</summary><ul><li>Built with version 1.2.2.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 26763/56156 (47.7%)</li>
<li><details><summary>Failed: 84/56156 (0.1%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.device_init.create_instance_device_intentional_alloc_fail
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_storage_large
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_storage_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_uniform_large
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_uniform_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_view_storage_r8g8b8a8_unorm
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_view_uniform_r8g8b8a8_unorm
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.command_pool
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.command_pool_transient
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.compute_pipeline
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_pool
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_pool_free_descriptor_set
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_set_layout_empty
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_set_layout_single
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device_group
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device_memory_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.event
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.fence
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.fence_signaled
Stack trace: Fail
...
```
</p></details></li>

<li>Skipped: 29309/56156 (52.2%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 26763/56156 (47.7%)</li>
<li><details><summary>Failed: 84/56156 (0.1%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.device_init.create_instance_device_intentional_alloc_fail
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_storage_large
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_storage_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_uniform_large
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_uniform_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_view_storage_r8g8b8a8_unorm
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_view_uniform_r8g8b8a8_unorm
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.command_pool
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.command_pool_transient
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.compute_pipeline
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_pool
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_pool_free_descriptor_set
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_set_layout_empty
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_set_layout_single
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device_group
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device_memory_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.event
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.fence
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.fence_signaled
Stack trace: Fail
...
```
</p></details></li>

<li>Skipped: 29309/56156 (52.2%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 25352/56233 (45.1%)</li>
<li><details><summary>Failed: 84/56233 (0.1%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.device_init.create_instance_device_intentional_alloc_fail
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_storage_large
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_storage_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_uniform_large
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_uniform_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_view_storage_r8g8b8a8_unorm
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_view_uniform_r8g8b8a8_unorm
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.command_pool
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.command_pool_transient
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.compute_pipeline
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_pool
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_pool_free_descriptor_set
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_set_layout_empty
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_set_layout_single
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device_group
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device_memory_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.event
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.fence
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.fence_signaled
Stack trace: Fail
...
```
</p></details></li>

<li>Skipped: 30797/56233 (54.8%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 26778/56156 (47.7%)</li>
<li><details><summary>Failed: 84/56156 (0.1%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.device_init.create_instance_device_intentional_alloc_fail
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_storage_large
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_storage_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_uniform_large
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_uniform_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_view_storage_r8g8b8a8_unorm
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_view_uniform_r8g8b8a8_unorm
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.command_pool
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.command_pool_transient
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.compute_pipeline
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_pool
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_pool_free_descriptor_set
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_set_layout_empty
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_set_layout_single
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device_group
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device_memory_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.event
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.fence
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.fence_signaled
Stack trace: Fail
...
```
</p></details></li>

<li>Skipped: 29294/56156 (52.2%)</li>
</ul></details>


</ul></details>

**Comment by trenouf** [comment]
Jenkins failures are the usual, as pointed out by Jax in another PR.

But I no longer seem to have the privilege to merge a PR with failing tests.

**Comment by trenouf** [comment]
It also won't let me direct push.

**Comment by trenouf** [comment]
I have merged this change in our internal repository.

**Comment by JaxLinAMD** [comment]
@trenouf  can you see bellow words above the button "Rebase and merge"?
"As an administrator, you may still merge this pull request."
if you can , you should have the permission to merge it.

**Pull request merged**